Sources/libcoverage/coverage.c

```diff
@@ -144,19 +175,104 @@ int cov_evaluate_crash(struct cov_context* context)
     int num_new_edges = internal_evaluate(context, context->crash_bits, &new_edges);
     free(new_edges.edges);
     return num_new_edges > 0;
-}
+ }
 
-int cov_compare_equal(struct cov_context* context, uint32_t* edges, uint64_t num_edges)
+int cov_compare_equal(struct cov_context* context, uint64_t* edges, uint64_t num_edges)
 {
     for (int i = 0; i < num_edges; i++) {
         int idx = edges[i];
         if (edge(context->shmem->edges, idx) == 0)
             return 0;
     }
-
     return 1;
 }
 
 void cov_clear_bitmap(struct cov_context* context) {
     memset(context->shmem->edges, 0, context->bitmap_size);
 }
+
+int uint64t_comp(const void *l, const void *r) {
+    const uint64_t lval = *(const uint64_t *)l;
+    const uint64_t rval = *(const uint64_t *)r;
+    
+    if(lval < rval){
+        return -1;
+    } else if (lval > rval){
+        return 1;
+    } else if (rval == lval) {
+        return 0;
+    }
+    return 0;
+}
+
+// Builds a list of the least visted edges
+// desiredEdgeCount is the number of edges to grab
+// expectedRounds is the number of rounds that the fuzzer expects to run on each sample. This ensures that samples with poor success rates aren't hit every time
+// Context is the libcoverage context
+// sorted_edge_count_list is a list of the count of each edge
+// sorted_edge_count_len is the length of sorted_edge_count_list
+// edges_ptr will point to malloced result array
+// num_edges is the length of edges_ptr
+int get_least_used_indicies(uint64_t desiredEdgeCount, uint64_t expectedRounds, struct cov_context* context, uint64_t * sorted_edge_count_list, uint64_t sorted_edge_count_len, uint64_t ** edges_ptr, uint64_t * num_edges) {
```

Comment:
I'm wondering if all of this could happen in the MarkovCorpus implementation since it seems to be specific to that? In that case, the CoverageEvaluator would optionally export the raw edge counts array which the MarkovCorpus would then use to compute these sets etc.